### DISCUSSION TITLE
study_link 모음

### DISCUSSION THREAD

**Discussion by hye-jj** [O-per/cakd3_Project3#4] [Study & Ideas]
### GAN
Introduction to Generative Adversarial Networks (GANs)
- https://learnopencv.com/introduction-to-generative-adversarial-networks/

Generative Adversarial Networks: Build Your First Models
- https://realpython.com/generative-adversarial-networks/

### Pytorch 
Collection of generative models in Pytorch version.
- https://pythonrepo.com/repo/znxlwm-pytorch-generative-model-collections

파이토치 튜토리얼 사이트 정리
- https://learning-sarah.tistory.com/entry/%ED%8C%8C%EC%9D%B4%ED%86%A0%EC%B9%98-%ED%8A%9C%ED%86%A0%EB%A6%AC%EC%96%BC-%EC%82%AC%EC%9D%B4%ED%8A%B8-%EC%A0%95%EB%A6%AC?category=849773

### Transformer, Bert
[파이썬]Transformer로 오피스 챗봇 만들기 - 이론편
- https://velog.io/@seolini43/%ED%8C%8C%EC%9D%B4%EC%8D%ACTransformer%EB%A1%9C-%EC%98%A4%ED%94%BC%EC%8A%A4-%EC%B1%97%EB%B4%87-%EB%A7%8C%EB%93%A4%EA%B8%B0-%EC%9D%B4%EB%A1%A0%ED%8E%B8

**Comment by hye-jj** [comment]
### YOLOv5
YOLOv5 학습 예제 (튜토리얼)  - 마스크 쓰고 있는/ 안 쓴 얼굴 찾기
- https://lv99.tistory.com/72

YOLOv5 모델로 Object detection 추론 - 미드 office parkour 예제 동영상
- https://lv99.tistory.com/73


[YOLO] Python과 OpenCV를 이용한 실시간 객체 탐지 알고리즘 구현
- https://deep-eye.tistory.com/6


[Python] OpenCV 웹캠 연결 문제 cv2.VideoCapture 해결 방법 #1
- https://deep-eye.tistory.com/73

yolo를 사용하여 물체 인식해보기
- https://webnautes.tistory.com/1406

파이썬 YOLOv5 Custom dataset 으로 학습하기
- https://yong0810.tistory.com/29

기능: 볼라드 탐지하기 (Bollard Detector) - Tiny YOLO 앱에 탑재하기(2단계)
- https://conkjh032.tistory.com/5


object detection 모델(pytorch)을 안드로이드 앱에 적용해보자 !
- https://developeryoung.tistory.com/4

YOLO 모델을 이용한 Object Detection 구현/Darkflow활용/Android적용
- https://ppowerppush.tistory.com/m/3

4부. CoreML을 이용하여 iOS에 YOLO v3 모델 탑재하기
- https://nero.devstory.co.kr/post/pj-too-real-04/

The iOS app
You can find the code in the YOLO folder.
- https://machinethink.net/blog/object-detection-with-yolo/

Android에서 내가 학습한 YOLO 모델을 이용해 Object Detection 구현하기
- https://junyoung-jamong.github.io/machine/learning/2019/01/25/Android%EC%97%90%EC%84%9C-%EB%82%B4-YOLO%EB%AA%A8%EB%8D%B8-%EC%82%AC%EC%9A%A9%ED%95%98%EA%B8%B0.html

윈도우 환경에서 YOLO로 실시간 객체탐지
- https://diy-project.tistory.com/124

머신러닝으로 자동차 object detection을 해보자 - (2) yolov2 tiny 모델을 COLAB에서 customize 하기
- https://developeryoung.tistory.com/2?category=879762

인공지능 알고리즘 yolov5s를 배워보자!
- https://blog.naver.com/kangtaekyu52/222434058536


### 어플구현
[Python] Python과 Java에서의 소켓 통신
- https://nowonbun.tistory.com/672

파이썬 소켓 프로그래밍 / 서버 프로그래밍 
- https://bagineer.tistory.com/3?category=825691

[Android] Android 와 Python 간 TCP/IP 소켓 통신 (AsyncTask 화면갱신)
- https://kurkurjae.tistory.com/53

[Python] 파이썬 앱 개발 모듈 키비(kivy)
- https://blog.naver.com/dsz08082/221842582052

[10]Android Yolo detection 구현하기
- https://sj-d.tistory.com/17

**Reply by hye-jj** [reply]
UIBezierPath
A path that consists of straight and curved line segments that you can render in your custom views.
- https://developer.apple.com/documentation/uikit/uibezierpath

**Reply by hye-jj** [reply]
darkflow
Translate darknet to tensorflow. Load trained weights, retrain/fine-tune using tensorflow, export constant graph def to mobile devices
- https://github.com/thtrieu/darkflow

**Comment by hye-jj** [comment]
### json to yolo
convert2Yolo 소개
convert2Yolo는 각종 datasets들을 YOLO[You Only Look Once]저자가 만든 darknet 프레임워크가 사용하는 label format으로 변경해주는 프로젝트입니다.

yolo에서 제공하는 데이터셋이 아닌 다른 데이터셋을 사용하려면 직접 파싱코드를 작성하는 방법, convert2Yolo에 해당 데이터셋을 파싱하는 코드를 작성, 새로운 툴을 사용하는 방법으로 이를 해결해야 합니다.
- https://deepbaksuvision.github.io/Modu_ObjectDetection/posts/02_02_Convert2Yolo.html
- https://github.com/ssaru/convert2Yolo


JSON 주석을 YOLO 형식으로 변환합니다.
- https://github.com/ultralytics/JSON2YOLO
- https://opensourcelibs.com/lib/json2yolo

특정 형식의 json 레이블 파일을 yolo를 기반으로 하는 txt 레이블 파일로 어떻게 변환합니까?
- https://stackoverflow.com/questions/68172777/how-do-i-convert-a-label-file-of-a-particular-format-of-json-to-txt-label-files

Bounding Box 좌표 값인 (x1, y2, x2, y2) 형식을
Yolo Style인 이미지 크기에 대한 비율 값으로 바꾸고, (centerX, centerY, w, h) 형식으로 바꾸는 소스코드
- https://eehoeskrap.tistory.com/357

**Comment by sumunoh** [comment]
bart 논문 리뷰
https://dladustn95.github.io/nlp/BART_paper_review/
